### ISSUE TITLE
Convert Lua bindings to flux_msg_t

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#224]
Lua bindings should be converted to use the `flux_msg_t` abstraction. Because of change in object destruction for `flux_msg_t` vs `zmsg_t` there _may_ be some memory leaks in the `zmsg_info_t` object handling. Still need to fully investigate.

Furthermore, the existence of the `zmsg_info_t` object was as a container for copies of `zmsg_t`'s due to awkward destroy-on-send behavior of zmsgs. Now that this is remedied via `flux_msg_t` it may be simpler for Lua bindings to pass around `flux_msg_t` as userdata and remove the `zmsg-lua.c`.

Either way, usage of raw `zmsg_t` should probably be cleaned up at some point in Lua code.

**Comment by grondo** [comment]
Oops, I may be confused. I guess Lua bindings still use old-style `flux_json_respond` which is based on deprecated `flux_sendmsg`, which still destroys the `zmsg` arg:

```C
int flux_sendmsg (flux_t h, zmsg_t **zmsg)
{       
    if (flux_send (h, *zmsg, 0) < 0)
        return -1;
    zmsg_destroy (zmsg);
    return 0;
}   
```

I guess if there is a mem leak it isn't new. :frowning:

I think I'll leave this issue open though because `zmsg-lua.c` should be replaced with something simpler based on `flux_msg_t` anyway.

**Comment by grondo** [comment]
closed by 78560c81e884f4c7ba93c5ac88f82cfce30dbce0
